swap_meet/vendor.py

```diff
@@ -1,2 +1,167 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    """A class representing a vendor who wants to swap items with other vendors
+
+    Attributes:
+        inventory (list): a list of items to swap with other vendors.
+    """
+
+    def __init__(self, inventory = None):
+        """Initializes the vendor with given inventory or set to None as default.
+
+        Args:
+            inventory(list): a list of items            
+        """
+        self.inventory = [] if inventory is None else inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        """Swaps items between the current vendor and another vendor.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap items with.
+            my_item (Item): The item the current vendor wants to trade.
+            their_item (Item): The item the other vendor wants to trade.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+
+        """
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.remove(my_item)
+        other_vendor.remove(their_item)
+
+        other_vendor.add(my_item)
+        self.add(their_item)
+
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        """Swaps the first item of the current vendor's inventory with the first item of the other vendor's inventory.
+
+        Parameters:
+            other_vendor (Vendor): The vendor to swap with.
+
+        Returns:
+            bool: True if the swap is successful, otherwise False.
+        """
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_item = self.inventory[0]
+        their_item = other_vendor.inventory[0]
+
+        return self.swap_items(other_vendor, my_item, their_item)
```

Comment:
One interesting point to note is that the README never specifies that the items need to be added to the end of each list! The tests don't check for a specific location for each item either. With this in mind, we know we are grabbing the first item from each list, so we could make this a true swap at position 0! One option would be to just reassign self.inventory[0] to first_item_other_vendor and other_Vendor.inventory[0] to first_item_vendor`.

We could also use a Python trick called multiple assignment. This allows us to assign multiple variables to new values all in one line. This might look like:

```suggestion
        self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]

        return True
```
This technically creates a tuple out of the list retrievals on the right side of the equal sign and then unpacks the values into the two variables on the left! This can be a neat trick to assign multiple variables at once!